### PR TITLE
Fixes #35072 - Save search props into store

### PR DESCRIPTION
**Pull request by ofedoren**
Someone with stronger React/Redux knowledge, please tell me if it can be solved in a "proper" way, since this looks more like a workaround.

Nevertheless, it fixes the described issue. Tested with https://github.com/theforeman/foreman_webhooks/pull/51 as well.

### CODE REVIEW

**Comment by adamruzicka** [comment]
Definitely not someone with stronger react/redux knowledge, but it does fix the issue observed in webhooks.

**Comment by MariaAga** [comment]
We cant delete this as it causes other issues (example: no autocomplete in this page: `/filters/new?role_id=` since it changes the url in the state for each resource).
a work around can be done like this probably:
https://github.com/theforeman/foreman/pull/9186/files#diff-d2bf4e51d3cc2f35e20cd77f4c38701181a410989333cb751da46b8cc756c15aR52
Anyway the after this ^ pr is merged I have a pr to delete the autocomplete which might also solve the bug that is in this pr

**Comment by ofedoren** [comment]
Thanks, @MariaAga, I've found a different workaround which works for /models page as well as for https://github.com/theforeman/foreman_webhooks/pull/51.

File: webpack/assets/javascripts/react_app/components/SearchBar/SearchBar.js [hunk 1]
```diff
@@ -17,6 +19,11 @@ const SearchBar = props => {
     onBookmarkClick,
     setAutocompleteSearchQuery,
   } = props;
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(updateController(controller, autocomplete.url, 'searchBar'));
+  }, [dispatch]);
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/SearchBar/SearchBar.js]: Can you add `controller, autocomplete.url` to this array?

**Review by MariaAga — APPROVED** [review]
Thanks! Fixes the issue in the models page and doesn't seem to break other things :tada: 
Test failures arent related

**Comment by stejskalleos** [comment]
@ofedoren @MariaAga This PR broke some integration tests, see [#35173](https://projects.theforeman.org/issues/35173) for the details

**Pull request merged**